hack/aws-acceptance-test-cleanup/main.go

```diff
@@ -759,6 +764,111 @@ func destroyBackoff(ctx context.Context, resourceKind string, resourceID string,
 	}, backoff.WithContext(expoBackoff, ctx))
 }
 
+func cleanupIAMRoles(ctx context.Context, iamClient *iam.IAM) error {
+	var rolesToDelete []*iam.Role
+	rolePrefix := "consul-k8s-"
+	err := iamClient.ListRolesPagesWithContext(ctx, &iam.ListRolesInput{},
+		func(page *iam.ListRolesOutput, lastPage bool) bool {
+			return filterRolesWithPrefix(page, lastPage, &rolesToDelete, rolePrefix)
+		})
+	if err != nil {
+		return fmt.Errorf("failed to list roles: %v", err)
+	}
+
+	if len(rolesToDelete) == 0 {
+		fmt.Println("Found no iamRoles to clean up")
+		return nil
+	}
```

Comment:
#nit might be useful to log how many roles we're attempting to clean up in the `else` case